src/main/java/de/qabel/desktop/storage/StorageDownload.java

```diff
@@ -1,16 +1,23 @@
 package de.qabel.desktop.storage;
 
+import java.io.Closeable;
+import java.io.IOException;
 import java.io.InputStream;
 
-public class StorageDownload {
+public class StorageDownload implements Closeable {
 	private InputStream inputStream;
 	private String mHash;
 	private long size;
+	private Closeable closeable;
```

Comment:
For whatever you download (but in the current implementations it's a CloseableHTTPResponse ... that needs to be closed after all contents have been fetched ... or soon enough to not fill up the connection pool